core/src/test/java/org/apache/iceberg/view/TestViewVersionParser.java

```diff
@@ -129,4 +129,19 @@ public void missingDefaultCatalog() {
         .isInstanceOf(IllegalArgumentException.class)
         .hasMessage("Cannot parse missing field: default-namespace");
   }
+
+  @Test
+  public void invalidRepresentations() {
+    String invalidRepresentations =
+        "{\"version-id\":1, \"timestamp-ms\":12345, \"schema-id\":1, \"summary\":{\"user\":\"some-user\"}, \"representations\": 23, \"default-namespace\":[\"one\",\"two\"]}";
+    assertThatThrownBy(() -> ViewVersionParser.fromJson(invalidRepresentations))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse representations from non-array: 23");
+
+    String missingRepresentations =
+        "{\"version-id\":1, \"timestamp-ms\":12345, \"schema-id\":1, \"summary\":{\"user\":\"some-user\"}, \"default-namespace\":[\"one\",\"two\"]}";
+    assertThatThrownBy(() -> ViewVersionParser.fromJson(missingRepresentations))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing field: representations");
```

Comment:
Nit: Missing the representations field seems like a different case and maybe should be a different test method.